crates/sparrow-instructions/src/evaluators/cast.rs

```diff
@@ -46,6 +50,17 @@ impl CastEvaluator {
                 ),
                 ty,
             ) if CastEvaluator::is_supported_interval_cast(ty) => true,
+            // Support casting from a duration to a duration with a more granular time unit
+            // However, arrow doesn't currently support this directly, so this hacks it a bit.
+            (DataType::Duration(TimeUnit::Second), DataType::Duration(_)) => true,
```

Comment:
For casting, it probably makes sense to allow both ways. We have a separate "can implicitly cast" which is the thing that should be limited to "only make it more granular". So here, this can just be -- if they're both durations then we can cast".

Reply:
Added the implicit cast